allofplos/article_class.py

```diff
@@ -676,47 +706,50 @@ def get_contributors_info(self):
-        """Whether the JATS article type is a correction or retraction.
+    def amendment_bool(self):
+        """Whether the JATS article type is a correction, retraction, or expression of concern.
 
+        These are the three article types ('amendments') that potentially warrant a change in the original article
+        that they reference (i.e., the 'related-doi'.)
         See https://jats.nlm.nih.gov/archiving/tag-library/1.1/attribute/article-type.html
-        :returns: True if a correction or retraction, False if not
+        :returns: True if an amendment article type, False if not
         :rtype: {bool}
         """
-        if self.type_ == 'correction' or self.type_ == 'retraction':
-            self._correct_or_retract = True
+        if self.type_ in ['correction', 'retraction', 'expression-of-concern']:
+            self._amendment = True
         else:
-            self._correct_or_retract = False
-        return self._correct_or_retract
-
-    def get_related_doi(self):
-        """For an article file, get the DOI of the first related article.
-
-        More strict in tag search if article is correction type
-        Use primarily to map correction and retraction notifications to articles that have been corrected
-        NOTE: what to do if more than one related article?
-        :return: first doi at that xpath location
+            self._amendment = False
+        return self._amendment
+
+    def get_related_dois(self):
+        """For a given article, get the list of DOIs of related PLOS articles.
+        Creates a dictionary of related dois & their type from the <related-articles> xpath location
+        Use primarily to map amendment notifications to articles that have been amended
+        :return: dictionary of related DOIs
+        :rtype: dict
         """
         related_article_elements = self.get_element_xpath(tag_path_elements=["/",
                                                                              "article",
                                                                              "front",
                                                                              "article-meta",
                                                                              "related-article"])
-        related_article = ''
-        if self.type_ == 'correction':
-            for element in related_article_elements:
-                if element.attrib['related-article-type'] in ('corrected-article', 'companion'):
-                    corrected_doi = element.attrib['{http://www.w3.org/1999/xlink}href']
-                    related_article = corrected_doi.lstrip('info:doi/')
-                    break
-        else:
-            try:
-                related_article_element = related_article_elements[0].attrib
-                related_article = related_article_element['{http://www.w3.org/1999/xlink}href']
+        related_article_dict = {}
+
+        try:
+            for elem in related_article_elements:
```

Comment:
Does this need to be in a try except block? If there are no elements in related_article_elements and all the other indices are on dictionaries, you can use the `.get` syntax and avoid any `IndexError`s.